### ISSUE TITLE
Show stem pro doesn't show up

### ISSUE THREAD

**Issue by boomwooq** [Mark2Mark/show-stems-pro-plugin#11]
Hi, sorry to bring this up again. 
It just happened today that Show Stem Pro doesn't show, so I think it cannot be activated for some reason. 
I haven't changed anything or updated anything recently. 

I'm on 3.3 (3316) 

![CleanShot 2024-08-19 at 15 33 47](https://github.com/user-attachments/assets/15933efa-0536-4f1d-9865-13085666cbb3)

**Comment by boomwooq** [comment]
More demo video. I've tried to uninstall and re-install already — doesn't work
![CleanShot 2024-08-19 at 15 46 37](https://github.com/user-attachments/assets/b900a31c-f4ab-4b88-9eb5-8d61cd601c57)

**Comment by florianpircher** [comment]
Might this be an issues with the reporter palette plugin? I see in the first screenshot that Stems Pro 3 is shown with a closed eye there.

**Comment by boomwooq** [comment]
I tried to open that eye, but it also didn't work—meanwhile, the rest of the plug-in works just fine. However, to reduce the factor, I decided to remove the Reporter plug-in. Unfortunately, it doesn't change anything.

**Comment by Mark2Mark** [comment]
The Eye symbol in the Reporter Toggler palette is only to show your favourite reporters in that palette. When you hit the "Edit" button, it will hide all the reporters which have the eye closed and only show those with the open eye.

If the plugin shows at all in the Reporter palette it is at least there.

I have not changed anything since quite a while, so my first instinct is to blame Glyphs (once more) for having changed something that affects this plugin. I can be wrong with that assumption though.

@boomwooq Can you launch Glyphs from the Terminal and see if we get some info when you try to activate the plugin? 👇

Here is how you launch Glyphs from the Terminal, to see if we get something logged, even if Glyphs crashes right after start.

### Long version:

1. (Quit and) Open Terminal
2. (Quit GlyphsApp if open)
3. Right click on GlyphsApp icon in Finder > Show Package Contents.
4. In there, go into the "MacOS" folder, and drop the contained binary file (black icon) on the Terminal Dock Icon.
5. Terminal now outputs the Glyphs’s console logs.

### Short version:
paste this into Terminal: `/Applications/Glyphs\ 3.app/Contents/MacOS/Glyphs\ 3` and hit enter.

**Comment by boomwooq** [comment]
Thank you Mark for your promtly replied! 
Here is the info:
`2024-08-19 19:52:38.465 Glyphs 3[2315:106454] __import 1
2024-08-19 19:52:38.465 Glyphs 3[2315:106454] __import 2
2024-08-19 19:52:38.478 Glyphs 3[2315:106454] __import 3
2024-08-19 19:52:38.478 Glyphs 3[2315:106454] __import 4
2024-08-19 19:52:38.503 Glyphs 3[2315:106454] 'Presenter' Paddle Bundle: 4.4.0 (4.4.0)
2024-08-19 19:52:38.507 Glyphs 3[2315:106454] 'Stems Pro 3' Paddle Bundle: 4.4.0 (4.4.0)
2024-08-19 19:52:38.549 Glyphs 3[2315:106454] 'Variable Font Preview 3' Paddle Bundle: 4.4.0 (4.4.0)
2024-08-19 19:52:39.143 Glyphs 3[2315:106454] Number of MIDI sources: 0
2024-08-19 19:52:39.145 Glyphs 3[2315:106454] Glyphs listening on port com.GeorgSeifert.Glyphs3
2024-08-19 19:52:39.145 Glyphs 3[2315:106454] Glyphs listening on port com.GeorgSeifert.Glyphs3.3316
2024-08-19 19:52:39.899 Glyphs 3[2315:106454] WARNING: Secure coding is not enabled for restorable state! Enable secure coding by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState: and returning YES.`

**Comment by Mark2Mark** [comment]
Thank you. And it does not log anything more when you try to activate Show Stems?

**Comment by boomwooq** [comment]
Nothing was showing up when I was trying to activate the plug-in. However, here is what shows up after opening the font file. In case it helps
```
2024-08-20 15:57:06.359 Glyphs 3[9490:771107] CFSocketSetAddress bind failure: 48
2024-08-20 15:57:06.359 Glyphs 3[9490:771107] HTTPServer error: Error Domain=HTTPServerError Code=0 "Unable to bind socket to address." UserInfo={NSLocalizedDescription=Unable to bind socket to address.}
2024-08-20 15:57:06.649 Glyphs 3[9490:771107] WARNING: Application performed a reentrant operation in its NSTableView delegate. This warning will become an assert in the future.
```

**Comment by Mark2Mark** [comment]
Thanks once more. Unfortunately nothing of those is related to this plugin.

Do you happen to have multiple versions of Glyphs on your Mac? If so, can you try an older version.

Find all glyphs versions [here](florianpircher.com/glyphs/versions/)


Note:
You can have several GlyphsApp versions installed.

**Comment by boomwooq** [comment]
I found the older version [here ](https://glyphsapp.com/learn/older-versions-of-glyphs)
Currently my version is 3.2.3 (3260)

And the plug-in is still not working. Do you have any recommended version that I should go for? 
When I activated the plug-in, it is still not showing anything. Here, just in case, it is from the terminal of the current version (mentioned above). 

```
/Applications/Glyphs\ 3\ copy.app/Contents/MacOS/Glyphs\ 3 
2024-08-20 21:48:19.492 Glyphs 3[11857:1012702] __import 1
2024-08-20 21:48:19.492 Glyphs 3[11857:1012702] __import 2
2024-08-20 21:48:19.506 Glyphs 3[11857:1012702] __import 3
2024-08-20 21:48:19.506 Glyphs 3[11857:1012702] __import 4
2024-08-20 21:48:19.731 Glyphs 3[11857:1012702] 'Presenter' Paddle Bundle: 4.4.0 (4.4.0)
2024-08-20 21:48:19.735 Glyphs 3[11857:1012702] 'Stems Pro 3' Paddle Bundle: 4.4.0 (4.4.0)
2024-08-20 21:48:19.777 Glyphs 3[11857:1012702] 'Variable Font Preview 3' Paddle Bundle: 4.4.0 (4.4.0)
2024-08-20 21:48:20.361 Glyphs 3[11857:1012702] Number of MIDI sources: 0
2024-08-20 21:48:20.362 Glyphs 3[11857:1012702] Glyphs listening on port com.GeorgSeifert.Glyphs3
2024-08-20 21:48:20.362 Glyphs 3[11857:1012702] Glyphs listening on port com.GeorgSeifert.Glyphs3.3260
2024-08-20 21:48:21.136 Glyphs 3[11857:1012702] WARNING: Secure coding is not enabled for restorable state! Enable secure coding by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState: and returning YES.
2024-08-20 21:48:24.668 Glyphs 3[11857:1012702] WARNING: Application performed a reentrant operation in its NSTableView delegate. This warning will become an assert in the future.
```

**Comment by Mark2Mark** [comment]
Sorry about the link, there must have been a typo, so Github resolved it into an invalid url. I fixed the link, so you can save it for later. It is way more convenient than the guide from the Glyphs page :)

Thanks for the log, unfortunately there is still no hint for me that points to this plugin.

→ A version where Show Stems still works for me is build 3310

→ which OS are you on?

→ sorry for all the trouble!

**Comment by Mark2Mark** [comment]
Thanks Boom!
One more Q, do you have Sonoma on there since recently and if so did it work before that? I don’t have Sonoma yet installed for testing, and currently I cannot, due to the time the newborn eats up at home. So I hope I can pin it down by asking seemingly silly questions. Thanks for your patience!

**Comment by florianpircher** [comment]
I tested with Glyphs 3316 on Sonoma (macOS 14) and it works as expected, so I don’t think the app/OS version is affecting this issue. But I have no other theory just yet. What is most strange to me is in the second post, the vide shows that the checkmark is never set in the _View_ menu for the _Show Stems Pro 3_ menu item. So I think the plugin is never activated properly somehow, or it gets immediately deactivated.

**Comment by Mark2Mark** [comment]
Thanks for checking, Florian!

> So I think the plugin is never activated properly somehow, or it gets immediately deactivated.

Right, as I understood that is the core of the issue.

@boomwooq Could you try this:
1. Duplicate your Plugins folder
1. Rename this new, duplicated Plugins folder (for example to "Plugins-TEMP"). It will serve as your backup to restore later
1. Go into the original "Plugins" folder and delete all plugins except Show Stems
1. Restart Glyphs and try again to activate it

→ I am curious if there is some interference with other plugins. Seems odd, but we actually had such a thing happening once with Show Red Arrows and one of my plugins.

Then later you can delete the "Plugins" folder which has only this one plugin inside and rename the "Plugins-TEMP" folder back to "Plugins" (then you should be back to your former state.

**Comment by boomwooq** [comment]
yea, I think it could be the case. However, I did accord to what you've suggested. It's still not activated properly. Attached here to show its action. But it's ok — we can pause the investigation for now, nothing urgent ;-) 
![CleanShot 2024-08-21 at 17 16 42](https://github.com/user-attachments/assets/d9d5d53c-1fa2-4cdc-beea-e46c3e44c5d6)

**Comment by Mark2Mark** [comment]
Thanks anyway! Looks like it appears for a split second. I will think about it and come back. Sorry for the inconvenience, Boom!

**Comment by eweracs** [comment]
Since this issue is still marked as open, I wanted to chime in:

This issue happened to me with Variable Font Preview some time ago. If I remember correctly, it had to do with the Paddle framework. Reinstalling the plugin only replaced the plugin folder, not the Paddle framework folder. Remove that as well and then reinstall the plugin.

**Comment by boomwooq** [comment]
@eweracs Hey, many thanks for sharing! Good trick, I'll try!

**Comment by Mark2Mark** [comment]
Thanks @eweracs, yes that should do the trick at least.

**Comment by boomwooq** [comment]
@eweracs @Mark2Mark works now! Thanks for the tip!

**Comment by typeservices** [comment]
I just wanted to chime in that I've had the same issue in 3.4.1 (3436). I've tried uninstalling the plugin and paddle framework, reinstalled and the same issue persists.

**Comment by Mark2Mark** [comment]
Hi @typeservices 
Can you please send me an email so that I can help you debugging?

@boomwooq - for you the same offer is still on ;)

hello/at/markfromberg.com

**Comment by Mark2Mark** [comment]
bump @boomwooq & @typeservices 

Also i just pushed an update, maybe this helps already (even though that was tackling other features)
